internal/migration/precheck.go

```diff
@@ -230,21 +224,20 @@ func (c *precheckContext) checkMachines(ctx context.Context) error {
 	if err != nil {
 		return errors.Annotate(err, "retrieving machines")
 	}
-	modelPresenceContext := common.ModelPresenceContext{Presence: c.presence}
 	for _, machine := range machines {
 		if machine.Life() != state.Alive {
 			return errors.Errorf("machine %s is %s", machine.Id(), machine.Life())
 		}
 
 		if statusInfo, err := machine.InstanceStatus(); err != nil {
 			return errors.Annotatef(err, "retrieving machine %s instance status", machine.Id())
-		} else if statusInfo.Status != status.Running {
+		} else if !status.IsInstanceViable(statusInfo) {
```

Comment:
Hmmm. Existing logic but this seems unnecessary. We should just care that the juju agent is running. If the instance is allocating or whatever, the agent won't be there anyway, and if the instance has reported an error, that's fine so long as the agent itself is still ok

Reply:
yeah it's bonkers.